pywbem_mock/_cimrepository.py

```diff
@@ -2485,236 +2521,295 @@ def _pull_operations_disabled(self):
-
-        objects = [] if result is None else [x[2] for x in result[0][2]]
+        # Handle ReturnQueryResultClass maybe or make it exception
+        instances = self.ExecQuery(FilterQueryLanguage, FilterQuery,
+                                   namespace=namespace)
 
-        return self._open_response(objects, namespace,
-                                   'PullInstancesWithPath', **params)
+        return self._open_response(instances, namespace,
+                                   'PullInstancesWithPath',
+                                   OperationTimeout,
+                                   MaxObjectCount,
+                                   ContinueOnError)
 
     ####################################################################
     #
     #   PullInstancesWithPath server request
     #
     ####################################################################
 
-    def PullInstancesWithPath(self, **params):
+    def PullInstancesWithPath(self, EnumerationContext, MaxObjectCount):
         """
-        Implements WBEM server responder for
+        Implements CIM repository responder for
         :meth:`~pywbem.WBEMConnection.OpenPullInstancesWithPath`
         with data from the instance repository.
         """
         self._pull_operations_disabled()
         namespace = self.namespace
         return self._pull_response(namespace, 'PullInstancesWithPath',
-                                   **params)
+                                   EnumerationContext, MaxObjectCount)
 
     ####################################################################
     #
     #   PullInstancePaths server request
     #
     ####################################################################
 
-    def PullInstancePaths(self, **params):
+    def PullInstancePaths(self, EnumerationContext, MaxObjectCount):
         """
-        Implements WBEM server responder for
+        Implements CIM repository responder for
         :meth:`~pywbem.WBEMConnection.OpenPullInstancePaths`
         with data from the instance repository.
         """
         self._pull_operations_disabled()
         namespace = self.namespace
-        return self._pull_response(namespace, 'PullInstancePaths', **params)
+        return self._pull_response(namespace, 'PullInstancePaths',
+                                   EnumerationContext, MaxObjectCount)
 
     ####################################################################
     #
     #   PullInstances server request
     #
     ####################################################################
 
-    def PullInstances(self, **params):
+    def PullInstances(self, EnumerationContext, MaxObjectCount):
         """
-        Implements WBEM server responder for
+        Implements CIM repository responder for
         :meth:`~pywbem.WBEMConnection.OpenPullInstances`
         with data from the instance repository.
         """
         self._pull_operations_disabled()
         namespace = self.namespace
```

Comment:
See previous comment about `namespace = self.namespace`. Only this time, the namespace should be available in EnumerationContext.